### PR TITLE
Add validation to metadata

### PR DESCRIPTION
**Pull request by Wathsara**
This PR is to
Validate the metadata to stop known internal server errors.

### CODE REVIEW

File: components/org.wso2.carbon.identity.sp.metadata.saml2/src/main/java/org/wso2/carbon/identity/sp/metadata/saml2/util/Parser.java [hunk 1]
```diff
@@ -198,8 +198,21 @@ public SAMLSSOServiceProviderDO parse(String metadata, SAMLSSOServiceProviderDO
             List<RoleDescriptor> roleDescriptors = entityDescriptor.getRoleDescriptors();
             //TODO: handle when multiple role descriptors are available
             //assuming only one SPSSO is inside the entitydescripter
-            RoleDescriptor roleDescriptor = roleDescriptors.get(0);
-            SPSSODescriptor spssoDescriptor = (SPSSODescriptor) roleDescriptor;
+            RoleDescriptor roleDescriptor;
+            SPSSODescriptor spssoDescriptor;
+
+            if (roleDescriptors.size() > 0) {
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.sp.metadata.saml2/src/main/java/org/wso2/carbon/identity/sp/metadata/saml2/util/Parser.java]: shall we use CollectionUtils.isNotEmpty

**Review by emswbandara — APPROVED** [review]

File: components/org.wso2.carbon.identity.sp.metadata.saml2/src/main/java/org/wso2/carbon/identity/sp/metadata/saml2/util/Parser.java [hunk 2]
```diff
@@ -198,8 +198,21 @@ public SAMLSSOServiceProviderDO parse(String metadata, SAMLSSOServiceProviderDO
...
+                roleDescriptor = roleDescriptors.get(0);
+            } else {
+                throw new InvalidMetadataException("Role descriptor not found.");
+            }
+
+            if (roleDescriptor instanceof SPSSODescriptor) {
+                spssoDescriptor = (SPSSODescriptor) roleDescriptor;
+            } else {
+                throw new InvalidMetadataException("Invalid role descriptor class found.");
+            }
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.sp.metadata.saml2/src/main/java/org/wso2/carbon/identity/sp/metadata/saml2/util/Parser.java]:
```suggestion
            if (!(roleDescriptor instanceof SPSSODescriptor)) {
                throw new InvalidMetadataException("Invalid role descriptor class found.");
            } 
            spssoDescriptor = (SPSSODescriptor) roleDescriptor;
```

We can get rid of else blocks by using this pattern right? Use it in other places as well

File: components/org.wso2.carbon.identity.sp.metadata.saml2/src/main/java/org/wso2/carbon/identity/sp/metadata/saml2/util/Parser.java (hunk 1 above)
Comment (Wathsara) [reply components/org.wso2.carbon.identity.sp.metadata.saml2/src/main/java/org/wso2/carbon/identity/sp/metadata/saml2/util/Parser.java]: Validated using `CollectionUtils.isEmpty` in order to remove the else block.

File: components/org.wso2.carbon.identity.sp.metadata.saml2/src/main/java/org/wso2/carbon/identity/sp/metadata/saml2/util/Parser.java (hunk 2 above)
Comment (Wathsara) [reply components/org.wso2.carbon.identity.sp.metadata.saml2/src/main/java/org/wso2/carbon/identity/sp/metadata/saml2/util/Parser.java]: Fixed.

**Review by AnuradhaSK — APPROVED** [review]

**Review by emswbandara — APPROVED** [review]

**Pull request merged**